moleculemaker/mmli-job-manager
# feat: Added API endpoints to run MOLLI jobs

## Problem
JobManager API currently only runs CLEAN job image, but it could likely be used to run MOLLI as well

Relates to https://github.com/moleculemaker/molli-frontend/pull/7 and https://github.com/SEDenmarkLab/molli/pull/46

## Approach
* Allow caller to pass `image_repo` as a parameter to `kube.create_job` (defaults to clean job image)
* Change default job type from `cutout` to `clean`
* Replace job_monitor usage with KubeWatcher - watch for job-related k8s events
    * NOTE: This does not yet send emails after job is completed
* Added new API endpoints for MOLLI Submit / Status / Results similar to CLEAN's:
    * `POST /job/molli/{jobId}/results`: Submit a new MOLLI job
    * `GET /job/molli/{jobId}`: Return job metadata and status
    * `GET /job/molli/{jobId}/results`: Return output files from the MOLLI job
    * `GET /job/molli/{jobId}/results/{fileName}`: Return arbitrary output files
    * `GET /job/molli/{jobId}/results/log`: Return log messages from running/completed jobs (currently unused)

## How to Test
Prerequisites: Download and unzip these [MOLLI input CDXML files](https://github.com/moleculemaker/mmli-job-manager/files/11874223/molli-input-cdxml.zip)

1. Navigate to https://molli.frontend.staging.mmli1.ncsa.illinois.edu/configuration
2. (Optional) log in using the button at the top-right
3. Select cores file: `BOX_cores_test_1.cdxml` (see attachment)
4. Select substituents file: `BOX_substituents_test_1.cdxml` (see attachment)
5. Submit the job and wait
    * NOTE: This job can take up to ~20 minutes on our current hardware
6. Once the job is completed, you should see the results appear in the UI
